ra/ra.go

```diff
@@ -1414,16 +1418,6 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(ctx context.Context, ba
 
 	ch := &authz.Challenges[challengeIndex]
 
-	if response.Type != "" && ch.Type != response.Type {
```

Comment:
This change appears unrelated, or am I missing something?

Reply:
¯\_(ツ)_/¯, how does the stat referenced look? If we _have_ to I'm fine just nuking this but I'd rather we uncomment the return and enforce the behavior that you update the challenge that you are actually trying to update otherwise I feel like we are just encouraging sloppy client implementation.

Reply:
I'm going to back out this change. I don't feel strongly enough to dig.

Reply:
Restored in 19b404d